test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -2156,26 +2190,52 @@ static int traffic_mngr_suite_init(void)
 	if (ret > 0)
 		goto skip_tests;
 
+	odp_tm_egress_init(&egress);
+	egress.egress_kind = ODP_TM_EGRESS_PKT_IO;
+	egress.pktio = xmt_pktio;
+
+	/* Get TM capabilities */
+	ret = odp_tm_egress_capabilities(&egress_capa, &egress);
+	if (ret) {
+		ODPH_ERR("Failed to retrieve tm capabilities");
+		return ret;
+	}
+
+	/* Check for sufficient TM queues */
+	if (egress_capa.max_tm_queues < NUM_TM_QUEUES)
+		goto skip_tests;
+
+	/* Check for sufficient TM levels */
+	if (egress_capa.max_levels < NUM_LEVELS)
+		goto skip_tests;
+
+	for (j = 0; j < NUM_LEVELS; j++) {
+		/* Per node fanin */
+		if (egress_capa.per_level[j].max_fanin_per_node <
+		    FANIN_RATIO)
+			break;
+	}
+
+	if (j != NUM_LEVELS)
+		goto skip_tests;
+
+	/* Init tm capabilities with matching egress capa until tm is created */
+	tm_capabilities = egress_capa;
+
 	/* Fetch initial dynamic update capabilities, it will be updated
 	 * later after TM system is created.
 	 */
-	ret = odp_tm_capabilities(capabilities_array, MAX_CAPABILITIES);
```

Comment:
Now that this line is removed, I think the comment above the line should also be removed as the comment becomes unaccurate and not helpful.